pkg/auth/auth.go

```diff
@@ -244,7 +245,25 @@ func getUserAndPass(opts *LoginOptions, password, userFromAuthFile string) (user
 	}
 	if password == "" {
 		fmt.Fprint(opts.Stdout, "Password: ")
-		pass, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		// Save and restore the terminal status on interruptions to
+		// avoid that the terminal remains in echoless state.
+		// This is necessary as a workaround for https://github.com/golang/go/issues/31180
+		stdin := int(os.Stdin.Fd())
+		oldState, err := terminal.GetState(stdin)
+		if err != nil {
+			oldState = nil
+		}
+		sigch := make(chan os.Signal, 1)
```

Comment:
the channel must be closed in a defer call otherwise the goroutine will leak forever.

Reply:
Good catch. I reset the signal interrupt handler but forgot the channel.

Commit amended.